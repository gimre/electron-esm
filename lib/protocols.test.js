
const { handleCjs, handleEsm } = require( './protocols' )

describe( 'cjs handler', ( ) => {
    test( 'returns a Buffer', async ( ) => {
        const result = await handleCjs( 'fake-module' )
        expect( result.constructor ).toBe( Buffer )
    } )

    test( 'returns an ES module wrapper', async ( ) => {
        const result = await handleCjs( 'fake-module' )
        expect( result.toString( ) ).toMatchSnapshot( )
    } )
} )
