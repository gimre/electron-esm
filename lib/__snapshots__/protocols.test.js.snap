// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cjs handler returns an ES module wrapper 1`] = `
"
        const exports = require( 'fake-module' )
        export default exports.default || exports
    "
`;
